File: my-submissions/m1833.c
int comp(const void* a, const void* b) {
    return *((int*) a) - *((int*) b);
}

int maxIceCream(int* costs, int costsSize, int coins) {
    qsort(costs, costsSize, sizeof(int), comp);
    
    for (int i = 0; i < costsSize; i++) {
        if (coins < costs[i])
            return i;
        coins -= costs[i];
    }

    return costsSize;
}

File: my-submissions/m1833.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        bars = 0
        costs.sort(reverse=True)

        while costs :
            if coins < costs[-1] :
                return bars
            bars += 1
            coins -= costs.pop()

        return bars